File: src/test/java/cloud/cosmin/checklister/repo/ItemRepoTest.java
package cloud.cosmin.checklister.repo;

import cloud.cosmin.checklister.dao.ItemEntity;
import cloud.cosmin.checklister.dao.ListEntity;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.Optional;
import java.util.UUID;

import static org.junit.Assert.*;

@RunWith(SpringRunner.class)
@SpringBootTest
public class ItemRepoTest {
    @Autowired
    private ListRepo listRepo;

    @Autowired
    private ItemRepo itemRepo;

    @Test
    public void testCreateListAddItem() {
        ListEntity listEntity = new ListEntity();
        listEntity.setId(UUID.randomUUID());
        listEntity.setTitle("title");
        ListEntity list = listRepo.save(listEntity);

        ItemEntity newItem = new ItemEntity();
        newItem.setContent("test content");
        newItem.setList(list);
        newItem.setRank(0);
        ItemEntity savedEntity = itemRepo.save(newItem);

        Optional<ItemEntity> maybeItem = itemRepo.findById(savedEntity.getId());
        assertTrue(maybeItem.isPresent());

        ItemEntity item = maybeItem.get();
        assertEquals("test content", item.getContent());
        assertEquals("text/plain", item.getContentType());
        assertEquals(list.getId(), item.getList().getId());
        assertEquals(0, item.getRank());
        assertEquals("title", list.getTitle());

        Optional<ListEntity> maybeListWithItem = listRepo.findById(list.getId());
        assertTrue(maybeListWithItem.isPresent());

        ListEntity listWithItem = maybeListWithItem.get();
        assertEquals(1, listWithItem.getItems().size());
    }
}

File: src/test/kotlin/cloud/cosmin/checklister/repo/ItemRepoTest.kt
package cloud.cosmin.checklister.repo

import cloud.cosmin.checklister.dao.ItemEntity
import cloud.cosmin.checklister.dao.ListEntity
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Ignore
import org.junit.Test
import java.util.*

class ItemRepoTest {
    private val listRepo: ListRepo? = null
    private val itemRepo: ItemRepo? = null

    @Test
    @Ignore
    fun testCreateListAddItem() {
        val listEntity = ListEntity()
        listEntity.id = UUID.randomUUID()
        listEntity.title = "title"
        val list = listRepo!!.save(listEntity)

        val newItem = ItemEntity()
        newItem.content = "test content"
        newItem.list = list
        newItem.rank = 0
        val savedEntity = itemRepo!!.save(newItem)

        val maybeItem = itemRepo.findById(savedEntity.id!!)
        assertTrue(maybeItem.isPresent)

        val item = maybeItem.get()
        assertEquals("test content", item.content)
        assertEquals("text/plain", item.contentType)
        assertEquals(list.id, item.list!!.id)
        assertEquals(0, item.rank.toLong())
        assertEquals("title", list.title)

        val maybeListWithItem = listRepo.findById(list.id!!)
        assertTrue(maybeListWithItem.isPresent)

        val listWithItem = maybeListWithItem.get()
        assertEquals(1, listWithItem.items!!.size.toLong())
    }
}